libr/core/linux_heap_glibc.c

```diff
@@ -303,19 +336,45 @@ static void GH(print_arena_stats)(RCore *core, GHT m_arena, MallocState *main_ar
 }
 
 static bool GH(r_resolve_main_arena)(RCore *core, GHT *m_arena) {
-	if (!core || !core->dbg || !core->dbg->maps) {
-		return false;
+
+	r_return_val_if_fail (core && core->dbg && core->dbg->maps, false);
+
+	/*
+	if (core->dbg->main_arena_resolved) {
```

Comment:
uh?

Reply:
I think that if `main_arena_resolved` this function should set the right value in `*m_arena`, otherwise you are already relying on the caller having the right value.